test/e2e/pvc_test.go

```diff
@@ -0,0 +1,255 @@
+	"github.com/openshift/cluster-image-registry-operator/test/framework"
+)
+
+func testDefer(t *testing.T, client *framework.Clientset) {
+	if t.Failed() {
+		scList, err := client.StorageClasses().List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the storage classes: %s", err)
+		} else {
+			framework.DumpYAML(t, "storageclasses", scList)
+		}
+
+		pvList, err := client.PersistentVolumes().List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the persistent volumes: %s", err)
+		} else {
+			framework.DumpYAML(t, "persistentvolumes", pvList)
+		}
+
+		pvcList, err := client.PersistentVolumeClaims(imageregistryv1.ImageRegistryOperatorNamespace).List(metav1.ListOptions{})
+		if err != nil {
+			t.Logf("unable to dump the persistent volume claims: %s", err)
+		} else {
+			framework.DumpYAML(t, "persistentvolumeclaims", pvcList)
+		}
+	}
+	framework.MustRemoveImageRegistry(t, client)
+}
+
+func createPV(t *testing.T, storageClass string) error {
+	client := framework.MustNewClientset(t, nil)
+	name := ""
+
+	for i := 0; i < 100; i++ {
+		pvName := fmt.Sprintf("pv-%s", rand.String(64))
+		localPath := fmt.Sprintf("/tmp/%s", pvName)
+
+		if err := os.MkdirAll(localPath, 0777); err != nil {
+			if os.IsExist(err) {
+				continue
+			}
+			return err
+		}
+
+		pv := &corev1.PersistentVolume{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: pvName,
+			},
+			Spec: corev1.PersistentVolumeSpec{
+				Capacity: corev1.ResourceList{
+					corev1.ResourceStorage: resource.MustParse("100Gi"),
+				},
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					corev1.ReadWriteOnce,
+					corev1.ReadOnlyMany,
+					corev1.ReadWriteMany,
+				},
+				PersistentVolumeSource: corev1.PersistentVolumeSource{
+					HostPath: &corev1.HostPathVolumeSource{
+						Path: localPath,
+					},
+				},
+				StorageClassName: storageClass,
+			},
+		}
+
+		_, err := client.PersistentVolumes().Create(pv)
+		if err == nil {
+			t.Logf("PersistentVolume %s created", pvName)
+			name = pvName
+			break
+		}
+
+		if errors.IsAlreadyExists(err) {
+			continue
+		}
+
+		t.Logf("unable to create PersistentVolume %s: %s", pvName, err)
+	}
+
+	if name == "" {
+		return fmt.Errorf("unable to create PersistentVolume")
+	}
+
+	err := wait.Poll(1*time.Second, framework.AsyncOperationTimeout, func() (bool, error) {
+		pv, pvErr := client.PersistentVolumes().Get(name, metav1.GetOptions{})
+		return (pv != nil && pv.Status.Phase != corev1.VolumePending), pvErr
+	})
+
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createPVWithStorageClass(t *testing.T) error {
+	client := framework.MustNewClientset(t, nil)
+
+	storageClassList, err := client.StorageClasses().List(metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("unable to list storage classes: %s", err)
+	}
+
+	for _, storageClass := range storageClassList.Items {
+		if err := createPV(t, storageClass.Name); err != nil {
```

Comment:
since a storageclass is required to have a provisioner, why do we need to do this?

shouldn't we only need to create PVs manually if there is no storageclass/provisioner defined on the cluster?

Reply:
> since a storageclass is required to have a provisioner, why do we need to do this?

The provisioner did not create PV. I have to create them manually.

> shouldn't we only need to create PVs manually if there is no storageclass/provisioner defined on the cluster?

@bparees I tried different configurations and only this works.

Reply:
> The provisioner did not create PV. I have to create them manually.

did you talk to the storage team about it to understand why it didn't do so?

> @bparees I tried different configurations and only this works.

so just creating a PV and then a PVC to claim it, does not work?  That seems very surprising, i'm pretty sure we do exactly that elsewhere.  I'd like you to discuss this with the storage team to understand why you had to do it this way, unless you understand it yourself.

Reply:
You don't have to learn everything, just enough to answer my questions and convince yourself+me that we are doing this the right/expected way.  

Right now it sounds like you found a way that works(and that's probably sufficient for what we specifically need right now), but you don't know why it works/has to be done that way/is correct.  I'd like to get to the bottom of that, not just for our own purposes, but because my one of my understandings(perhaps incorrect) at a product level is that we should have PV autoprovisioning that works out of the box on an AWS cluster.  I think it's a useful exercise.

Reply:
> but you don't know why it works/has to be done that way/is correct.

I need to know how our test cluster was configured and I will answer to you. I am almost 100% sure that we do not have a autoprovisioner. But for now I can not prove it.